resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesDriverBuilder.scala

```diff
@@ -43,12 +43,12 @@ private[spark] class KubernetesDriverBuilder {
       new DriverServiceFeatureStep(conf),
       new MountSecretsFeatureStep(conf),
       new EnvSecretsFeatureStep(conf),
-      new LocalDirsFeatureStep(conf),
       new MountVolumesFeatureStep(conf),
       new DriverCommandFeatureStep(conf),
       new HadoopConfDriverFeatureStep(conf),
       new KerberosConfDriverFeatureStep(conf),
-      new PodTemplateConfigMapStep(conf))
+      new PodTemplateConfigMapStep(conf),
+      new LocalDirsFeatureStep(conf))
```

Comment:
why not move this right after `new MountVolumesFeatureStep(conf)`?

Reply:
I think some volume mounts maybe setup in PodTemplateConfigMapStep, so I move to last to ensure that.

Reply:
Yep, since the intent is for users to provide custom local dir volumes via either mount volumes or pod templates it needs to appear after both

Reply:
Either is fine, just wanted to point out that the pod template is initialized before any step is executed (i.e. `PodTemplateConfigMapStep` is not where the template is loaded).